cpp/src/arrow/compute/exec/exec_plan.h

```diff
@@ -521,9 +521,9 @@ ARROW_EXPORT Result<std::vector<std::shared_ptr<RecordBatch>>> DeclarationToBatc
 ARROW_EXPORT Future<std::vector<std::shared_ptr<RecordBatch>>> DeclarationToBatchesAsync(
     Declaration declaration, ExecContext* exec_context = default_exec_context());
 
-/// \brief Utility method to run a declaration return results as a RecordBatchReader
-Result<std::unique_ptr<RecordBatchReader>> DeclarationToReader(Declaration declaration,
-                                                               bool use_threads);
+/// \brief Utility method to run a declaration and return results as a RecordBatchReader
+ARROW_EXPORT Result<std::unique_ptr<RecordBatchReader>> DeclarationToReader(
```

Comment:
I am curious why are we adding `ARROW_EXPORT` here?

Reply:
IIRC, not having this `ARROW_EXPORT` led to an MSVC build error (for reasons [Weston describes here](https://github.com/westonpace/arrow/pull/27#discussion_r1046558052)) already in the pre-PR code, and I just took the opportunity to fix it.